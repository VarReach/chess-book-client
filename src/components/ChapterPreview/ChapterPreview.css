
.chapter-preview {
  display: flex;
  align-items: center;
  height: 112px;
  position: relative;
  background-color: #f7f7f7;
  border: 1px solid #c0c0c0;
  margin-bottom: 20px;
  padding: 0 35px;
  box-shadow: 0 5px 0 0 #00000020;
}

.chapter-preview__chapter-details {
  max-width: 100%;
}

.chapter-preview__chapter-details a {
  display: inline-block;
  font-family: 'Libre Baskerville', serif;
  color: #242425;
  text-decoration: none;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  width: 100%;
}

.chapter-preview__chapter-details a:hover {
  text-decoration: underline;
}

.chapter-preview__chapter-details h2 {
  margin: 3px 0;
  font-size: 30px;
}

.chapter-preview__chapter-index {
  font-size: 14px;
  font-weight: 500;
  color: #48484a;
}

.chapter-preview__completion-indicator {
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  transform: rotate(-90deg);
  letter-spacing: 0.8px;
  left: -67px;
  height: 20px;
  width: 112px;
  top: 45px;
  background-color: #479226;
  box-shadow: -5px 0 0 0 #00000020;
  color: #f7f7f7;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 10px;
}

.chapter-preview__completed-date {
  position: absolute;
  right: 40px;
  display: flex;
  align-items: center;
  height: 100%;
  font-size: 14px;
  font-weight: 500;
  color: #838383;
}

@media only screen and (max-width: 1300px) {
  .chapter-preview__chapter-details h2 {
    font-size: 24px;
  }

  .chapter-preview__chapter-index {
    font-size: 12px;
  }

  .chapter-preview__completion-indicator {
    height: 10px;
    left: -62px;
    top: 50px;
  }

  .chapter-preview__completion-indicator span {
    display: none;
  }

  .chapter-preview {
    padding: 0 20px;
  }
}

@media only screen and (max-width: 600px) {
  .chapter-preview__chapter-details h2 {
    font-size: 18px;
  }

  .chapter-preview__chapter-index {
    font-size: 12px;
  }

  .chapter-preview__completed-date {
    display: none;
  }
}
