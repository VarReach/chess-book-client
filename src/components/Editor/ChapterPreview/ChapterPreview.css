.editor-chapter__button-holder button { 
  width: 30px;
  height: 30px;
  font-size: 16px;
  margin-bottom: 10px;
}

button.info-button {
  position: absolute;
  left: -10px;
  font-size: 10px;
  width: 20px;
  height: 20px;
}

.editor__chapter button {
  position: relative;
  padding: 0;
  padding-top: 3px;
  outline: none;
  background-color: #f7f7f7;
  border: 1px solid #c0c0c0;
  box-shadow: 0 3px 0 0 #d9d9d9;
}

.editor__chapter button:hover {
  cursor: pointer;
}

.editor__chapter button:hover, .editor__chapter button:focus {
  outline: 3px solid #00000014;
  box-shadow: none;
}

.editor__chapter h3 {
  margin: 3px 0 0;
  font-size: 30px;
  display: inline-block;
  font-family: 'Libre Baskerville', serif;
  color: #242425;
  width: 100%;
}
.editor__chapter h3:hover {
  color: #990D35;
}

.editor__chapter {
  display: flex;
  align-items: center;
  justify-content: space-between;
  min-height: 112px;
  position: relative;
  background-color: #f7f7f7;
  border: 1px solid #c0c0c0;
  text-decoration: none !important;
  margin-bottom: 20px;
  padding: 5px 5px 5px 15px;
  box-shadow: 0 3px 0 0 #d9d9d9;
}

.editor__chapter:hover, .editor__chapter:focus {
  box-shadow: none;
  outline: 3px solid #d9d9d9;
}

.editor-chapter-info {
  color: #f7f7f7;
  padding: 20px;
  left: 55px;
}

.editor-chapter-info:after {
	right: 100%;
	top: calc(50% - 16px);
	border: solid transparent;
	content: " ";
	height: 0;
	width: 0;
	position: absolute;
	pointer-events: none;
	border-color: #00000000;
  border-right-color: #242831;
  /* border-width: 0 16px 8px 16px; */
  border-width: 16px 8px 16px 16px;
}

.editor-chapter__info {
  display: flex;
  align-items: center;
}

.editor-chapter__button-holder {
  display: flex;
  flex-direction: column;
}


@media only screen and (max-width: 1300px) {
  .editor__chapter h3 {
    font-size: 24px;
  }
}

@media only screen and (max-width: 600px) {
  .editor__chapter h3 {
    font-size: 18px;
  }
}