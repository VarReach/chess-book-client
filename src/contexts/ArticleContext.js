import React, { Component } from 'react'

const nullArticle = {
  title: null,
  content: [],
}

const ArticleContext = React.createContext({
  totalArticles: null,
  article: nullArticle,
  error: null,
  setArticle: () => {},
  clearArticle: () => {},
  setError: () => {},
})

export default ArticleContext;

export class ArticleProvider extends Component {
  constructor(props) {
    super(props);
    this.state = {
      totalArticles: null,
      article: nullArticle,
      error: null,
      setArticle: this.setArticle,
      clearArticle: this.clearArticle,
      setError: this.setError
    };
  }

  setArticle = (resp) => {
    this.setState({ 
      article: {
        title: resp.title,
        content: resp.content,
      },
      totalArticles: resp.total_articles,
    });
  }

  clearArticle = () => {
    this.setState({
      article: nullArticle,
      totalArticles: null,
    })
  }

  setError = (error) => {
    this.setState({ error });
  }

  render() {
    return (
      <ArticleContext.Provider value={this.state}>
        {this.props.children}
      </ArticleContext.Provider>
    )
  }
}
