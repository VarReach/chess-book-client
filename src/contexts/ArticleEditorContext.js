import React, { Component } from 'react'

const nullArticle = {
  id: null,
  chapter_id: null,
  title: null,
  content: {},
  date_created: null,
  date_published: null,
  date_modified: null,
}

const ArticleEditorContext = React.createContext({
  article: {},
  error: null,
  setArticle: () => {},
  clearArticle: () => {},
  setError: () => {},
})

export default ArticleEditorContext;

export class ArticleEditorProvider extends Component {
  constructor(props) {
    super(props);
    this.state = {
      article: {},
      error: null,
      setArticle: this.setArticle,
      setError: this.setError,
    };
  }

  setChapters = (article) => {
    this.setState({ chapters: resp.chapters, totalChapters: resp.totalChapters });
  }

  setError = (error) => {
    this.setState({ error });
  }

  render() {
    return (
      <ArticleEditorContext.Provider value={this.state}>
        {this.props.children}
      </ArticleEditorContext.Provider>
    )
  }
}
