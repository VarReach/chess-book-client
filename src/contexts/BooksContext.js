import React, { Component } from 'react'

const BooksContext = React.createContext({
  bookId: null,
  books: {},
  error: null,
  setBooks: () => {},
  setBookId: () => {},
  setError: () => {},
});

export default BooksContext;

export class BooksProvider extends Component {
  constructor(props) {
    super(props);
    this.state = {
      bookId: null,
      books: {},
      error: null,
      setBooks: this.setBooks,
      setBookId: this.setBookId,
      setError: this.setError
    };
  }

  setBooks = (books, bookId, fn) => {
    this.setState({ books }, () => {
      this.setBookId(bookId, fn);
      fn && fn();
    });
  }

  setBookId = (bookId) => {
    this.setState({ bookId });
  }

  setError = (error) => {
    this.setState({ error });
  }

  render() {
    return (
      <BooksContext.Provider value={this.state}>
        {this.props.children}
      </BooksContext.Provider>
    )
  }
}
