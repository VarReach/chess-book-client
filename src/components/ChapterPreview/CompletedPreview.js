import React from 'react';
import './ChapterPreview.css';

function CompletedPreview(props){
  return (
    <>
      <div className="chapter-preview__completion-indicator">
        <span>completed</span>
      </div>
      <div className="chapter-preview__completed-date">
        <span >
          {props.date}
        </span>
      </div>
    </>
  )
}

export default CompletedPreview;