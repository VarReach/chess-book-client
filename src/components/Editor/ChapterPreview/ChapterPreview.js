import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { Draggable } from 'react-beautiful-dnd';
import './ChapterPreview.css';
import EditorBookContext from '../../../contexts/EditorBookContext';

class Chapter extends Component {
  static contextType = EditorBookContext;

  handleDeleteChapter = (e) => {
    e.preventDefault();
    this.props.handleDeleteChapter(this.props.chapter.id);
  }

  render() {
    return (
      <Draggable draggableId={this.props.chapter.id} index={this.props.index}>
        {(provided) => (
          <div className="editor__chapter"
            {...provided.draggableProps}
            {...provided.dragHandleProps} //apply this to whatever you want the user to click on to drag it. Can be just a small section
            ref={provided.innerRef}
          >
            <Link to={`/editor/chapter/${this.props.chapter.id}`}><h3>{this.props.chapter.title}</h3></Link>
            <button onClick={this.handleDeleteChapter}>Delete</button>
          </div>
        )}
      </Draggable>
    )
  }
}

export default Chapter