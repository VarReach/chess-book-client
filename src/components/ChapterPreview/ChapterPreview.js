import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import CompletedPreview from './CompletedPreview';
import './ChapterPreview.css';
import BooksContext from '../../contexts/BooksContext';

class ChapterPreview extends Component {
  static contextType = BooksContext;

  render() {
    const { chapter, index } = this.props;
    const bookId = this.context.bookId;
    return (
      <Link to={`/book/${bookId}/chapter/${index+1}`} className={'chapter-preview' + (this.props.date ? ' completed-chapter' : '')}>
        {this.props.date && <CompletedPreview chapterId={chapter.id} date={this.props.date}/>}
        <div className="chapter-preview__chapter-details">
          <span className="chapter-preview__chapter-index">{`Chapter ${index+1}`}</span>
          <h2>{chapter && chapter.title}</h2>
        </div>
      </Link>
    );
  }
}

export default ChapterPreview