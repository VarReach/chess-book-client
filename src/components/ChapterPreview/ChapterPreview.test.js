import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ChapterPreview from './ChapterPreview';
import { BooksProvider } from '../../contexts/BooksContext';
import { MemoryRouter } from 'react-router-dom';

describe(`NotefulForm component`, () => {
  const props = {
    chapter: {
      title: 'Chapter title',
    },
    index: 1,
  };

  it('renders a ChapterPreview by default', () => {
    const wrapper = shallow(<ChapterPreview />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders the ChapterPreview given props', () => {
    const wrapper = shallow(<ChapterPreview {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})