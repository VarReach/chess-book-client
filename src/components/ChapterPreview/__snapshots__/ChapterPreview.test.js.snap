// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotefulForm component renders a ChapterPreview by default 1`] = `
<Link
  className="chapter-preview"
  replace={false}
  to="/book/undefined/chapter/NaN"
>
  <div
    className="chapter-preview__chapter-details"
  >
    <span
      className="chapter-preview__chapter-index"
    >
      Chapter NaN
    </span>
    <h2 />
  </div>
</Link>
`;

exports[`NotefulForm component renders the ChapterPreview given props 1`] = `
<Link
  className="chapter-preview"
  replace={false}
  to="/book/undefined/chapter/2"
>
  <div
    className="chapter-preview__chapter-details"
  >
    <span
      className="chapter-preview__chapter-index"
    >
      Chapter 2
    </span>
    <h2>
      Chapter title
    </h2>
  </div>
</Link>
`;
